### PR TITLE
style(frontend): new ButtonAsLink for the footer [GIX-3049]

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

We want to change the style of the icons in the footer. That is actually a version of the `tertiary` button. So we adapt the code to use it.

# Changes

- Create a type for the button styles and use it in the button components.
- Change the style of `tertiary` button class to have a generic style and only specific after link.
- Create `ButtonAsLink` component that is basically the same as `Button` but with <a>.
- Use `ButtonAsLink` in the `Footer`.
- Unrelated: we adjust the footer spacing.

# Tests

Only changed the footer. The rest is the same.

Screenshot 2024-10-08 at 13 31 19

### CODE REVIEW

File: src/frontend/src/lib/components/core/Footer.svelte [hunk 1]
```diff
@@ -13,31 +14,34 @@
...
 	<div class="flex flex-row items-center gap-4">
 		<a
 			href={OISY_REPO_URL}
 			rel="external noopener noreferrer"
 			target="_blank"
-			class="rounded-full bg-white p-2"
 			aria-label={$i18n.navigation.text.source_code_on_github}
 		>
-			<IconGitHub />
+			<ButtonIcon ariaLabel={$i18n.navigation.text.source_code_on_github}>
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/core/Footer.svelte:29]: Mmmh a `button` within a `a`. Is that semantically speaking correct?

File: src/frontend/src/lib/components/core/Footer.svelte [hunk 2]
```diff
@@ -13,31 +14,34 @@
...
+				{$i18n.navigation.text.source_code_on_github}
+			</ButtonIcon>
 		</a>
 
 		<a
 			href={OISY_TWITTER_URL}
 			rel="external noopener noreferrer"
 			target="_blank"
-			class="rounded-full bg-white p-2"
 			aria-label={$i18n.navigation.text.source_code_on_github}
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/core/Footer.svelte:39]: GitHub? Shouldn't be Twitter?

File: src/frontend/src/lib/components/core/Footer.svelte [hunk 3]
```diff
@@ -13,31 +14,34 @@
...
 		>
-			<IconTwitter />
+			<ButtonIcon>
+				<IconTwitter slot="icon" />
+				{$i18n.navigation.text.source_code_on_github}
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/core/Footer.svelte:43]: Same same

File: src/frontend/src/lib/components/core/Footer.svelte [hunk 4]
```diff
@@ -13,31 +14,34 @@
...
 
 		<a
 			href={OISY_TWITTER_URL}
 			rel="external noopener noreferrer"
 			target="_blank"
-			class="rounded-full bg-white p-2"
 			aria-label={$i18n.navigation.text.source_code_on_github}
 		>
-			<IconTwitter />
+			<ButtonIcon>
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/core/Footer.svelte:41]: Same same

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/core/Footer.svelte:41]: As above

File: src/frontend/src/lib/components/core/Footer.svelte (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/core/Footer.svelte:29]: I was in doubt a bit about this, but we will have some button styles that will be applied all over, even to link <a> elements.

So, I am not sure how to tackle this. The other solutions is to include such classes for both <button> and <a>, but I don't really like the idea somehow

Comment (peterpeterparker) [reply src/frontend/src/lib/components/core/Footer.svelte:29]: I should be a link. I think that from an accessibility and semantic point of view, using a button here, moreover stricly for styling purpose, is incorrect.

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/core/Footer.svelte]: Right, check last commit please: I created a new component for that, WDYT?

File: src/frontend/src/lib/components/ui/ButtonAsLink.svelte [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<script lang="ts">
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/ui/ButtonAsLink.svelte:1]: I suggest to rename this component to `ExternalLinkIcon`.

File: src/frontend/src/lib/components/ui/ButtonIcon.svelte [hunk 6]
```diff
@@ -1,11 +1,14 @@
 <script lang="ts">
+	import type { ButtonColorStyle } from '$lib/types/style';
+
 	export let button: HTMLButtonElement | undefined = undefined;
+	export let colorStyle: ButtonColorStyle = 'tertiary';
 	export let testId: string | undefined = undefined;
-	export let ariaLabel: string;
+	export let ariaLabel: string | undefined = undefined;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/ui/ButtonIcon.svelte:7]: Do we need this change?

File: src/frontend/src/lib/i18n/en.json [hunk 7]
```diff
@@ -41,7 +41,8 @@
 			"more_settings": "More settings",
 			"menu": "Your wallet address, settings, sign-out and external links",
 			"changelog": "Open the changelog of $oisy_name on GitHub to review the latest updates",
-			"submit_ticket": "Open the ticket submission form on GitHub to report an issue or request a feature"
+			"submit_ticket": "Open the ticket submission form on GitHub to report an issue or request a feature",
```
Comment (peterpeterparker) [inline src/frontend/src/lib/i18n/en.json:44]: Can be simplified: "Report an issue or request a feature on GitHub"

File: src/frontend/src/lib/i18n/en.json [hunk 8]
```diff
@@ -41,7 +41,8 @@
...
+			"open_twitter": "Open the X (Twitter) account of $oisy_name"
```
Comment (peterpeterparker) [inline src/frontend/src/lib/i18n/en.json:45]: nitpick: Oisy does not have a Twitter account per sé, the foundation has. "Open the DFINITY X/Twitter feed"

File: src/frontend/src/lib/components/ui/ButtonIcon.svelte (hunk 6 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/ui/ButtonIcon.svelte]: forgot to remove, removed

File: src/frontend/src/lib/i18n/en.json (hunk 7 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/i18n/en.json]: changed

File: src/frontend/src/lib/i18n/en.json (hunk 8 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/i18n/en.json]: right!

File: src/frontend/src/lib/components/ui/ButtonAsLink.svelte (hunk 5 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/ui/ButtonAsLink.svelte]: Yes! Make sense, changed

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

**Pull request merged**